modules/apps/journal/journal-web/src/main/resources/META-INF/resources/js/UndoRedo.js

```diff
@@ -0,0 +1,214 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {ClayButtonWithIcon} from '@clayui/button';
+import React, {useCallback, useEffect, useState} from 'react';
+
+const META_FIELD_NAMES = {
+	description: 'descriptionMapAsXML',
+	friendlyURL: 'friendlyURL',
+	title: 'titleMapAsXML',
+};
+
+export default function UndoRedo({
+	initialDefaultLanguageId,
+	languageId,
+	namespace,
+}) {
+	const [
+		// eslint-disable-next-line no-unused-vars
+		{defaultLanguageId, history, selectedLanguageId, step},
+		setState,
+	] = useState({
```

Comment:
Let's use the useState pattern for this `[x, setX] = useState({..}) `
I'm not sure it the name "state" is the appropiate, but right now I have no better suggestions so let's keep it as 
`const [state, setState] = useState({...})`

Reply:
I would rather keep the state destructured as it is more readable as well as easier to access the fields.